internal/ocm/ocm.go

```diff
@@ -22,6 +22,9 @@ type ClusterServiceClientSpec interface {
 	// GetClusterStatus sends a GET request to fetch a cluster's status from Cluster Service.
 	GetClusterStatus(ctx context.Context, internalID InternalID) (*arohcpv1alpha1.ClusterStatus, error)
 
+	// GetClusterInflightChecks sends a GET request to fetch a cluster's inflight checks from Cluster Service.
+	GetClusterInflightChecks(ctx context.Context, internalID InternalID) (*arohcpv1alpha1.InflightCheckList, error)
```

Comment:
Might Cluster Service ever paginate the inflight check list?  Since we moved to Go 1.23 I've been building [iterators](https://pkg.go.dev/iter) around CS list types to encapsulate pagination.  See [/internal/ocm/iterators.go](https://github.com/Azure/ARO-HCP/blob/main/internal/ocm/iterators.go).  I don't know if it would be worth it in this case, but just wanted to mention it for consideration.

Reply:
Let me take a closer look at the CS backend.
The number of inflights won't exceed 10-20 inflights soon, and with relatively small struct per inflight, I doubt we'll use paging for this endpoint.

Reply:
After double-checking, the default page size in CS is 100, so in the case of inflights, it will be one page.
I don't see a scenario where the number of inflights reaches that number.